cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -641,20 +641,29 @@ func sglogBranches(key string, branches []zoekt.RepositoryBranch) sglog.Field {
 }
 
 func (s *Server) indexArgs(opts IndexOptions) *indexArgs {
-	parallelism := math.Ceil(float64(s.CPUCount) / float64(s.IndexConcurrency))
+	cpuCount := s.cpuCount(opts)
+	parallelism := math.Ceil(float64(cpuCount) / float64(s.IndexConcurrency))
 	return &indexArgs{
 		IndexOptions: opts,
-
 		IndexDir:    s.IndexDir,
 		Parallelism: int(parallelism),
-
 		Incremental: true,
 
 		// 1 MB; match https://sourcegraph.sgdev.org/github.com/sourcegraph/sourcegraph/-/blob/cmd/symbols/internal/symbols/search.go#L22
 		FileLimit: 1 << 20,
 	}
 }
 
+// cpuCount consults both the server flags and index options to determine the number
+// of CPUs to use for indexing. If the index option is provided, it always overrides
+// the server flag.
+func (s *Server) cpuCount(opts IndexOptions) int {
```

Comment:
minor: maybe just make this return a float64. That would simplify the min logic below and at the only callsite you cast to float64.